app/models/embedded_ansible_worker/runner.rb

```diff
@@ -71,14 +71,12 @@ def provider_url
     server = MiqServer.my_server(true)
 
     if MiqEnvironment::Command.is_container?
-      host = ENV["ANSIBLE_SERVICE_HOST"]
-      path = "/api/v1"
+      URI::HTTPS.build(:host => ENV["ANSIBLE_SERVICE_HOST"], :path => "/api/v1").to_s
+    elsif Rails.env.development?
+      URI::HTTP.build(:host => "localhost", :path => "/api/v1", :port => 54321).to_s
     else
-      host = server.hostname || server.ipaddress
-      path = "/ansibleapi/v1"
+      URI::HTTPS.build(:host => server.hostname || server.ipaddress, :path => "/ansibleapi/v1").to_s
     end
-
-    URI::HTTPS.build(:host => host, :path => path).to_s
```

Comment:
Minor, but you can also code this up such that each path in the conditional creates a hash, with `:scheme` as one of the keys, then you can DRY up the URI, build, to_s bits down here into a single line.